app/controllers/close_contacts_controller.rb

```diff
@@ -42,4 +42,15 @@ def update
                                created_by: current_user.email,
                                comment: "User edited a close contact (ID: #{cc.id}).")
   end
+
+  # Delete an existing close contact record
+  def destroy
+    redirect_to root_url unless current_user.can_create_patient_close_contacts?
+    cc = current_user.get_patient(params.permit(:patient_id)[:patient_id])&.close_contacts.find_by(id: params.permit(:id)[:id])
+    cc.update(archived: true)
+    cc.save
```

Comment:
is this even necessary after calling `update`?

Reply:
I'm not sure, but that's what's used in the update function above, so I followed that idiom. I'll check once static analysis passes.